packages/flutter/lib/src/material/switch.dart

```diff
@@ -752,6 +725,18 @@ class _MaterialSwitchState extends State<_MaterialSwitch> with TickerProviderSta
   @override
   bool? get value => widget.value;
 
+  void updateCurve() {
```

Comment:
It looks like the curve and reverseCurve properties should be Switch and SwitchTheme parameters (could be a separate PR).  They'll be a little complicated to explain, but they are clearly needed to configure the switch for different look-and-feels.